WikiWatershed/model-my-watershed
# Set BiG-CZ Mode by Host or Query Parameter

## Overview

Previously we were setting BiG-CZ specific settings based on an environment variable. However, since in live deployment both MMW and BiG-CZ will be running off the same (virtual) servers, we cannot rely on environment variables to distinguish them.

This approach looks for a `?bigcz=true` query parameter, or the **Host** header of the request coming from `[*.]portal.bigcz.org`. If the request is coming from that host, or that query parameter is passed, we enable BiG-CZ settings. Host inspection will be used in production, and query parameter is to support development.

Also tagging @mmcfarland and @hectcastro for code review.

Connects #1904 
Supersedes #1953 

## Testing Instructions

* Check out this branch
 * Go to [:8000/?bigcz=true](http://localhost:8000/?bigcz=true) and type in `clientSettings.data_catalog_enabled` in the console. Ensure its value is `true`.
 * Go to [:8000](http://localhost:8000/) without the query parameter, and ensure that `data_catalog_enabled` is `false`
* Try to spoof the `Host` header to `portal.bigcz.org` or `staging.portal.bigcz.org` and ensure that `data_catalog_enabled` is `true`
